tests/functions/test_deconcatenate_column.py

```diff
@@ -3,13 +3,32 @@
 
 @pytest.mark.functions
 def test_deconcatenate_column(dataframe):
-    df = dataframe.concatenate_columns(
+    df_orig = dataframe.concatenate_columns(
         column_names=["a", "decorated-elephant"],
         sep="-",
         new_column_name="index",
     )
-    df = df.deconcatenate_column(
-        column_name="index", new_column_names=["A", "B"], sep="-"
+    index_original = list(df_orig.columns).index("index")
+    index_next = index_original + 1
+    df = df_orig.deconcatenate_column(
+        column_name="index",
+        new_column_names=["col1", "col2"],
+        sep="-",
+        preserve_position=False,
+    )
+    assert "col1" in df.columns
+    assert "col2" in df.columns
+    # Test for `preserve_position` kwarg
+    df = df_orig.deconcatenate_column(
```

Comment:
I think the block from line 22 to 34 should be an independent test on its own. Could you break it out, and give it an informative test function name, e.g. `test_deconcatenate_column_preserve`?